hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/AbstractBatchEntitySelectFetchInitializer.java

```diff
@@ -82,10 +70,37 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 
 	protected abstract void registerResolutionListener();
 
+	protected void resolveKey(
+			RowProcessingState rowProcessingState,
+			ToOneAttributeMapping referencedModelPart,
+			FetchParentAccess parentAccess) {
+		if ( state != State.UNINITIALIZED ) {
+			return;
+		}
+		if ( !isAttributeAssignableToConcreteDescriptor( parentAccess, referencedModelPart ) ) {
+			return;
```

Comment:
I don't know how often this gets called, but I think it might be beneficial to set `state = State.MISSING;` before returning.